std/format.d

```diff
@@ -4362,14 +4362,44 @@ T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     return parse!T(input);
 }
 
+private T rawRead(T, Range)(ref Range input)
+    if (isSomeString!Range || ElementType!(Range).sizeof == 1)
+{
+    union X
+    {
+        ubyte[T.sizeof] raw;
+        T typed;
+    }
+    X x;
+    foreach (i; 0 .. T.sizeof)
+    {
+        static if (isSomeString!Range)
+        {
+            x.raw[i] = input[0];
+            input = input[1 .. $];
+        }
+        else
+        {
+            // TODO: recheck this
+            x.raw[i] = cast(ubyte) input.front;
+            input.popFront();
+        }
+    }
+    return x.typed;
+}
+
 /**
    Reads an integral value and returns it.
  */
 T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     if (isInputRange!Range && isIntegral!T && !is(T == enum))
 {
+
     import std.algorithm.searching : find;
     import std.conv : parse, text;
+
+    if (spec.spec == 'r') return rawRead!T(input);
```

Comment:
Following the logic before, here if somebody wants to parse a wide string with `%r` you would issue a run-time error:

```d
if (spec.spec == 'r')
{
    static if (is(Unqual!(ElementType!Range) == char) || is(Unqual!(ElementType!Range) == byte) || is(Unqual!(ElementType!Range) == ubyte))
        return rawRead!T(input);
    else
        throw new Exception("The raw read specifier %r may only be used with narrow strings and ranges of bytes."); 
}
```